rosette/rbl/rosette/space.rbl

```diff
@@ -312,7 +323,7 @@
 (defRMethod NameSpace (produce ctxt-tuple & production)
 (let* [
     [[ctxt _] ctxt-tuple]
-    [[channel unification_ptrn product] production]
+    [[channel unification_ptrn & products] production]
```

Comment:
This forces the products to be a tuple. The hello_world_again.rho no longer compiles because `['Rholang-world-477707192] != 'Rholang-world-477707192`.

Without the &, the hello_world_again.rho contract still compiles.

Reply:
I guess the issue is that now everything is a Tuple so we have to de-tuple (is that the right word?) the singles.

I guess one option is to change the helloworld contract itself as follows 

`for ([world] <- helloworld) { ... }`

And the other is to inject the following in the produce method
```

rosette> (defProc (detuple-if-single list) (if (= (size list) 1) (head list) list))
'detuple-if-single
rosette> (detuple-if-single [1 2 3])
[1 2 3]
rosette> (detuple-if-single [1])
1
```

where it gets injected like follows

```
    [[channel unification_ptrn & products] production]
    [products (detuple-if-single products)]
```

Reply:
Take a look now. Any read with multiple arguments was already expecting a tuple, so it makes sense to send one automatically in that case. We can now handle zero arguments as well. For uniformity, it seems best to have the single argument case be a tuple as well.